### PR TITLE
SDK-5657: Mute SDK

### PR DESCRIPTION
**Pull request by CTLalit**
…ic unmute API

- Refactor mute from relative 24h check to absolute expiry timestamp model
- Add X-WZRK-MUTE-DURATION response header support (absolute epoch ms from backend)
- Store mute expiry in new Long key (comms_mute_expiry_ts) replacing Int key (comms_mtd)
- Add migration logic in isMuted() to read old comms_mtd key and convert to new format
- Add setMuteExpiry(epochMs) for backend-provided mute durations
- Add unmute() public API on CleverTapAPI to programmatically clear mute state
- Add parseMuteExpiryMs() in NetworkManager to extract duration from response headers
- Retain backward compatibility: old SDKs ignore new header and use 24h default
- Mark getMuted() as deprecated in favor of getMuteExpiry()
- Add unit tests for migration, custom expiry, unmute, and edge cases

Constants.java:
  - Add KEY_MUTE_EXPIRY and DEFAULT_MUTE_DURATION_MS constants

CtApi.kt:
  - Add HEADER_MUTE_DURATION response header constant

NetworkRepo.kt:
  - isMuted(): now checks absolute expiry with migration fallback for old key
  - getMuteExpiry(): reads Long expiry from new storage key
  - setMuted(): stores now + 24h as absolute expiry
  - setMuteExpiry(): stores arbitrary absolute epoch ms expiry
  - unmute(): clears mute expiry to 0

NetworkManager.kt:
  - shouldMuteSdk(): extracts mute duration before calling setMuted
  - parseMuteExpiryMs(): parses X-WZRK-MUTE-DURATION header safely
  - setMuted(): accepts optional muteExpiryMs instead of Response object
  - unmute(): public method delegating to NetworkRepo

CleverTapAPI.java:
  - Add public unmute() method with Javadoc

NetworkRepoTest.kt:
  - Update existing mute tests for new expiry-based model
  - Add migration tests (active mute, expired mute, key precedence)
  - Add setMuteExpiry, unmute, and infinite mute tests

## Summary by CodeRabbit

* **New Features**
  * Added `unmute()` method to immediately clear backend-imposed mute state and resume normal network and event operations.

* **Improvements**
  * Enhanced mute handling to support backend-defined expiry timestamps, enabling more granular communication control independent of offline/online settings.

### CODE REVIEW

**Comment by francispereira** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/clevertap-platform/pr-checks/6b376851-6c56-412c-a71e-2e74b5cda80d?refs=6e3826ca3b1f2b93ba142c857c4cf453869dcb8a&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/clevertap-platform/pr-checks/6b376851-6c56-412c-a71e-2e74b5cda80d/license?refs=6e3826ca3b1f2b93ba142c857c4cf453869dcb8a&source=prChecksComment) |
| :white_check_mark: | Code Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/clevertap-platform/project/9a1fd2dd-3229-4564-b8c3-9660985efc7a/pr-check/6d3800d1-6e16-451c-bf53-ed0518fba363?refs=6e3826ca3b1f2b93ba142c857c4cf453869dcb8a&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt [hunk 1]
```diff
@@ -80,17 +80,48 @@ internal class NetworkRepo(
...
+            return now < muteExpiryMs
+        }
+        // Migration: check old key for an active mute from a previous SDK version
+        @Suppress("DEPRECATION")
+        val legacyMuteTs = getMuted()
+        if (legacyMuteTs > 0) {
+            val legacyExpiryMs = (legacyMuteTs + 24 * 60 * 60) * 1000L
+            if (now < legacyExpiryMs) {
+                // Migrate to new format so this path is only hit once
+                setMuteExpiry(legacyExpiryMs)
```
Comment (piyush-kukadiya) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt:100]: After migrating old `comms_mtd` to new `comms_mute_expiry_ts`, the old key is left with stale data. Should clear it?

File: clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt [hunk 2]
```diff
@@ -100,25 +131,39 @@ internal class NetworkRepo(
...
+            val expiryMs = clock.currentTimeMillis() + Constants.DEFAULT_MUTE_DURATION_MS
+            setMuteExpiry(expiryMs)
         } else {
-            StorageHelper.putInt(
-                context,
-                config.accountId,
-                Constants.KEY_MUTED,
-                0
-            )
+            setMuteExpiry(0L)
```
Comment (piyush-kukadiya) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt:143]: call `unmute()`?

**Review by piyush-kukadiya — COMMENTED** [review]
left comments

**Comment by Anush-Shand** [comment]
@coderabbitai review

**Review by Anush-Shand — APPROVED** [review]

File: clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt (hunk 1 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt:100]: keep this for downgrade scenario.

**Review by piyush-kukadiya — APPROVED** [review]

File: clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt (hunk 1 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt:100]: lets not remove this for downgrade case.

File: clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt (hunk 2 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkRepo.kt:143]: `unmute()` method can have more func hidden inside so setting explicitly.

**Review by piyush-kukadiya — APPROVED** [review]
